### PR TITLE
Update bootstrap to 4.5.0

### LINKED ISSUE #437: Update to latest bootstrap
We have neglected our bootstrap installation for way too long, we need to go ahead and update from 2.3 -> 4.

### PR DESCRIPTION
**Pull request by camdenmoors**
Closes #437 #529 
This

- Updates bootstrap to 4.5.0
- Stops ignoring CVE-2016-10735 CVE-2019-8331 in CI tests
- Fixes the navigation bar that selects between divisions in game summary

There is only one problem left that I can see but I am a bit stuck on how to fix. When putting items in :admin_menu they are put inside a div which will make it not show up in the navbar dropdown.

### CODE REVIEW

**Comment by rbclark** [comment]
Are there any well options that look good with bootstrap 4? I noticed that the well was missing in the footer on this branch:

Screen Shot 2020-07-14 at 10 46 43 PM
Screen Shot 2020-07-14 at 10 46 35 PM

**Comment by rbclark** [comment]
The legend is missing the line to the left of it, is there an option to restore this?

**Comment by rbclark** [comment]
The page does not shrink down to mobile size quite as well as it used to:

Screen Shot 2020-07-14 at 10 50 13 PM
Screen Shot 2020-07-14 at 10 50 38 PM

Specifically the large margin on the right. Filling out the page with the table all the time might look better? I'm not sure.

**Comment by rbclark** [comment]
Progress bars look to all be broken for me on the Game summary page.

Screen Shot 2020-07-14 at 10 52 24 PM

**Comment by camdenmoors** [comment]
@rbclark Should all be fixed now.

**Comment by rbclark** [comment]
Screen Shot 2020-07-15 at 9 46 38 AM

h3 muted has been replaced with `text-muted`

Screen Shot 2020-07-15 at 9 47 00 AM

**Comment by rbclark** [comment]
The height for the progress bar currently looks a little awkward, it was previously 20px however it seems that bootstrap made it skinnier. Can we set it back to 20px? https://getbootstrap.com/docs/4.0/components/progress/#height

**Comment by rbclark** [comment]
Registration page formatting seems to currently be broken

Screen Shot 2020-07-15 at 5 17 47 PM

Same for the login page & Create a Team page.

The Join a team page could use some help as well:

Screen Shot 2020-07-15 at 5 28 01 PM

Screen Shot 2020-07-15 at 5 28 42 PM

**Comment by rbclark** [comment]
Current tab on the navbar is not currently highlighted.

**Comment by rbclark** [comment]
Show/Hide button is currently broken on the Team Show and Team Summary pages.

**Comment by rbclark** [comment]
It wasn't as noticeable before, but team #5 and #6 below the show/hide button show as the same color in the striped table. I'm fine with just showing 6 teams instead of 5 if that is the easiest way to solve this.

Screen Shot 2020-07-15 at 5 39 05 PM

File: app/views/games/_team_list.html.haml [hunk 1]
```diff
@@ -13,13 +14,14 @@
       / Get the first teams score and use that as the full score bar
       - max_score = teams.first.current_score
       / We pass in starting_rank here to start the list at the proper index
-      = render partial: 'games/team_list_tbody', locals: { teams: teams.slice(0..4), starting_rank: 1, max_score: max_score }
+      = render partial: 'games/team_list_tbody', locals: { teams: teams.slice(0..5), starting_rank: 1, max_score: max_score, tbody_id: '' }
```
Comment (rbclark) [inline app/views/games/_team_list.html.haml:17]: Setting the `tbody_id` with a default value in the `games/team_list_tbody` avoids the `tbody_id` needing to be set to `''` here.

**Comment by rbclark** [comment]
Team summary solved challenges expansion has the same problem that the teams table had. Also for some reason expanding the solved challenges causes the app to jump to the top of the page.

File: app/views/users/registrations/_user_form.html.haml [hunk 2]
```diff
@@ -11,60 +11,56 @@
...
   -# if this is a new user registering then on_update_page should be false.
   - if on_update_page
     = render partial: 'users/registrations/update_user_form', locals: { f: f }
   .controls
-    = invisible_recaptcha_tags text: 'Submit', :class => "btn"
-- if on_update_page
-  = render partial: 'users/registrations/delete_user_form'
+    = invisible_recaptcha_tags text: 'Submit', :class => "btn btn-primary"
+//- if on_update_page
+//  = render partial: 'users/registrations/delete_user_form'
```
Comment (rbclark) [inline app/views/users/registrations/_user_form.html.haml:66]: Does this remove the ability for users to delete their account?

File: Gemfile [hunk 3]
```diff
@@ -5,14 +5,15 @@ source 'https://rubygems.org'
 gem 'activerecord-precounter'
 gem 'awesome_nested_fields'
 gem 'bootsnap', '>= 1.1.0', require: false
+gem 'bootstrap', '~> 4.5.0'
 gem 'bootstrap-kaminari-views'
```
[the remark this answers is not in the document]
Comment (camdenmoors) [reply Gemfile:9]: bootstrap
Seems that updating bootstrap brings a new font that I'm a fan of, along with breaking the navigation bar on smaller devices. Should be an easy migration

File: app/assets/stylesheets/custom-bootstrap.css.scss [hunk 4]
```diff
@@ -34,11 +34,8 @@ th.text-center {
   text-overflow: ellipsis;
 }
 
-.remove_top_padding {
-  margin-top:-20px;
-}
 .control-label-required {
-	@extend .control-label;
+	@extend .control-label !optional;
```
Comment (rbclark) [inline app/assets/stylesheets/custom-bootstrap.css.scss:38]: It looks like the `control-label` class may have been removed in bootstrap4, so this extend may just no longer be necessary: https://getbootstrap.com/docs/4.0/migration/#forms-1

File: app/assets/stylesheets/custom-bootstrap.css.scss [hunk 5]
```diff
@@ -261,13 +272,17 @@ td form {
 .noborderbottom {
 	border:none;
 }
+
 .well-small {
 	min-height:10px;
 }
```
Comment (rbclark) [inline app/assets/stylesheets/custom-bootstrap.css.scss:278]: It looks like parts of this file has the CSS indentation completely out of wack with other parts. If you could standardize it that would be great.

**Pull request merged**